pkg/client/client.go

```diff
@@ -215,3 +218,24 @@ func (c *Client) getHelmClient(releaseNamespace string) (helmclient.Client, erro
 
 	return helmClient, nil
 }
+
+// CreateOrUpdate attempts first to create the object given but if an AlreadyExists error
```

Comment:
My only worry about this is that we might end up silently updating accidentally leaked resources instead of creating new ones (like clusters for example). It can be prevented by adding an explicit check if the resource exists before calling to `CreateOrUpdate` but what is the benefit of having them together?

Reply:
In the tests we don't tend to deal with Kubernetes resources directly, we work with abstractions. We have a `Cluster` type that we configure and then build the kubernetes manifests from that. Because of this, we never have the `resourceVersion` of a resource so can't actually perform an update.

Reply:
Ah good point.